### PR TITLE
Allow anonymous package

### PR DESCRIPTION
**Pull request by borela**
## Description

Allow anonymous packages to be bootstrapped.

## Motivation and Context

It is often useful to create packages without name for functional testing where you may have many environments to test, naming these packages is unnecessary as they won't be published anywhere.

Fixes [#848](https://github.com/lerna/lerna/issues/848).

## How Has This Been Tested?

On a local repo, lerna almost supported this feature out of the box, all that I needed to do was filter the anonymous package from the commands.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by borela** [comment]
No problem, I'll add the tests.

**Comment by borela** [comment]
@evocateur Sorry for the delay, I added some tests and fixed some merge conflicts. The tests failing are the ones that were failing in the master before, I am not sure if it is related to my changes.

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "glob": "^7.1.2",
     "globby": "^6.1.0",
     "graceful-fs": "^4.1.11",
-    "inquirer": "^3.2.1",
+    "inquirer": "^3.1.0",
```
Comment (evocateur) [inline package.json]: I don't think you chose the right merge resolution here.

Comment (evocateur) [inline package.json]: Also, you should be using `yarn`, not npm 5. That's the reason your builds are failing.

File: test/fixtures/BootstrapCommand/tepid/packages/package-1/node_modules/external-1/package.json [hunk 2]
```diff
@@ -1,4 +0,0 @@
-{
-  "name": "external-1",
```
Comment (evocateur) [inline test/fixtures/BootstrapCommand/tepid/packages/package-1/node_modules/external-1/package.json]: These files definitely should not have been deleted.

File: test/fixtures/PublishCommand/anonymous/packages/anonoymous/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "dependencies": {
```
Comment (evocateur) [inline test/fixtures/PublishCommand/anonymous/packages/anonoymous/package.json]: None of these test fixtures need the explicit `node_modules` that were moved under this path.

Comment (borela) [inline test/fixtures/PublishCommand/anonymous/packages/anonoymous/package.json]: I actually don't know why the commits were so messed up, I fixed most issues, reverted the merge and sent the new commits.

**Comment by borela** [comment]
@evocateur Sorry for the delay , I hadn't time to fix the issues in this pull, would this still be accepted or does this feature is already implemented in newer versions?

**Comment by evocateur** [comment]
Yes, feel free to re-open when you have time, I was just trying to clear out stalled or out-of-date PRs.

**Pull request closed without merging**